src/main/java/io/HttpReaderWriter.java

```diff
@@ -15,8 +15,7 @@ public HttpReaderWriter(IMessageIO messageIO, IRequestBuilder builder, HttpRespo
 
   public HttpRequest getHttpRequest() throws IOException, InvalidHttpRequestException {
     socketIO.openClientConnection();
-    String message = getMessage();
-    return builder.createRequest(message);
+    return builder.createRequest(socketIO);
```

Comment:
I find it a little strange that the `HttpReaderWriter` no longer knows how to read the HTTP request and instead is delegating that responsibility to the `RequestBuilder`. Working with sockets seems like a pretty significant enough responsibility in its own right; adding it to "parsing the message from that socket" might be an SRP violation.